client/src/views/files/handler/viewer/FileViewer.vue

```diff
@@ -0,0 +1,77 @@
+<!-- Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS -->
+
+<template>
+  <file-handler @file-loaded="onFileLoaded">
+    <!-- file-viewer component -->
+    <component
+      :is="viewerComponent"
+      :content-info="contentInfoRef"
+    />
+  </file-handler>
+</template>
+
+<script setup lang="ts">
+import { ref, Ref, type Component, inject, shallowRef } from 'vue';
+import {
+  ImageViewer,
+  VideoViewer,
+  SpreadsheetViewer,
+  DocumentViewer,
+  AudioViewer,
+  TextViewer,
+  PdfViewer,
+} from '@/views/files/handler/viewer';
+import { Information, InformationLevel, InformationManager, InformationManagerKey, PresentationMode } from '@/services/informationManager';
+import { DetectedFileType, FileContentType } from '@/common/fileTypes';
+import { FileContentInfo } from '@/views/files/handler/viewer/utils';
+import FileHandler from '@/views/files/handler/FileHandler.vue';
+
+const informationManager: InformationManager = inject(InformationManagerKey)!;
+const viewerComponent: Ref<Component | null> = shallowRef(null);
+const contentInfoRef: Ref<FileContentInfo | undefined> = ref(undefined);
+
+const emits = defineEmits<{
+  (e: 'handlerReady'): void;
+}>();
+
+async function onFileLoaded(contentInfo: FileContentInfo, fileInfo: DetectedFileType): Promise<void> {
+  contentInfoRef.value = contentInfo;
+
+  const component = await getComponent(fileInfo);
+  if (!component) {
```

Comment:
What does the page display in this case? We show a toast, and what else? Since `handlerReady` will never be called, I'm guessing the state is a bit uncertain.